src/Map/src/MapFactory.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ *
+ * @internal
+ */
+final class MapFactory
+{
+    /**
+     * @param array{
+     *  center?: array{lat: float, lng: float},
+     *  zoom?: float,
+     *  markers?: list<array{
+     *    position: array{lat: float, lng: float},
+     *    title?: string,
+     *    infoWindow?: array{
+     *      content?: string,
+     *      headerContent?: string,
+     *    },
+     *  }>,
+     * } $array The array representation of the map
+     */
+    public static function fromArray(array $array): Map
```

Comment:
Do you think it can makes sense to use the Symfony OptionsResolver here, to replace the multiple checks on `$array` properties?

Reply:
At first, I did not want to bloat in dependencies, but this is the exact role of ConfigResolver so ... if you too think we should leverage it, i'm all in :)

Reply:
Alternatively we could add static functions to the verious DTO classes (Point, Marker, etc) ... which seems the best place to implement this..

wdyt ?

Reply:
> At first, I did not want to bloat in dependencies, but this is the exact role of OptionsResolver so ... if you too think we should leverage it, i'm all in :)

We can try with the OptionsResolver yes, but we need to be careful about performances I think.